scripts/create_deprecated_modules.lean

```diff
@@ -0,0 +1,173 @@
+/-
+Copyright (c) 2025 Damiano Testa. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Damiano Testa
+-/
+
+--import Mathlib.Init
+import Lean.Elab.Command
+import Mathlib.Tactic.Linter.DeprecatedModule
+-- a comment here to test `keepTrailing
+
+/-!
+#  Create a deprecated module
+
+This file defines the lean script for creating a deprecated module.
+-/
+
+open Lean Elab Command
+
+namespace DeprecatedModule
+
+/--
+`getHeader fname fileContent keepTrailing` takes as input two strings and a `Bool`ean.
+It uses
+* `fname`, as the path of a file (which need not exist);
+* `fileContent`, as the content of `fname` (regardless of whether the file exists and what its
+  content is);
+* `keepTrailing` a boolean to control whether to keep trailing comments.
+
+It returns the content of `fileContent` up to the final import, including trailing comments if `keepTrailing = true` (the command always trims trailing whitespace after the last comment).
+-/
+def getHeader (fname fileContent : String) (keepTrailing : Bool) : IO String := do
+  let (stx, _) ← Parser.parseHeader (Parser.mkInputContext fileContent fname)
+  let stx := if keepTrailing then stx.raw else stx.raw.unsetTrailing
+  let some substring := stx.getSubstring? | throw <| .userError "No substring: we have a problem!"
+  let upToAllImports : Substring := {substring with startPos := 0}
+  return upToAllImports.toString
+
+/--
+`getHeaderFromFileName fname keepTrailing` is similar to `getHeader`, except that it assumes that
+`fname` is the actual path of a file, and uses `fname`'s content as input to `getHeader`.
+-/
+def getHeaderFromFileName (fname : String) (keepTrailing : Bool) : IO String := do
+  getHeader fname (← IO.FS.readFile fname) keepTrailing
+
+/--
+`mkDeprecation customMessage` returns the formatted syntax
+`deprecated_module "customMessage" (since := "YYYY-MM-DD")`,
+where the date is today's date.
+-/
+def mkDeprecation (customMessage : String := "Auto-generated deprecation") :
+    CommandElabM Format := do
+  let msgStx := if customMessage.isEmpty then none else some <| Syntax.mkStrLit customMessage
+  let dateStx := Syntax.mkStrLit s!"{← Std.Time.PlainDate.now}"
```

Comment:
Is this date in the UTC time zone (or that does this depend on the computer executing it)?

Reply:
I think that it is, yes, but I have not done intensive testing.  I remember that the date-commands had some issues in CI, so I'm sure that there will be some joy involved in getting this to work on all setups.  This is also a reason to wait on adding the workflow to CI.

Reply:
One great feature of Lean is that we have access to the source (and it's in a language I can read): [my impression](https://github.com/leanprover/lean4/blob/81b85203c90439fb7d7e0cec32130d62d8cb7a2e/src/Std/Time/Zoned.lean#L19-L28) is that this uses the time zone on the computer. Oops. Let me ask on zulip about the best way forward.

Reply:
[Asked on zulip](https://leanprover.zulipchat.com/#narrow/channel/270676-lean4/topic/Current.20date.20in.20UTC/with/517985138)

Reply:
```suggestion
  -- Get the current date in UTC: we don't want this to depend on the user computer's time zone.
  let dateStx := Syntax.mkStrLit s!"{(← DateTime.now (tz := .UTC)).toPlainDate}"

Reply:
Thanks for chasing this up!